src/common/code_environment.cc

```diff
@@ -46,19 +46,18 @@ std::ostream &operator<<(std::ostream &oss, const enum code_environment_t e)
   return oss;
 }
 
-#if defined(HAVE_SYS_PRCTL_H) && defined(PR_GET_NAME) /* Since 2.6.11 */
+#if defined(HAVE_PTHREAD_GETNAME_NP)
```

Comment:
Could you please use the following instead?

`#if defined(HAVE_PTHREAD_GETNAME_NP) && !defined(_WIN32)`

As shown here [1], ``pthread_getname_np`` doesn't return the process name on Windows. It will only provide an explicitly assigned thread name.

[1]  https://github.com/ceph/ceph/blob/master/src/log/Log.cc#L407

Reply:
@petrutlucian94 sure. fixed and repushed.